File: chess.py
def lagya(xK, yK, xL, yL):
    return xK == xL or yK == yL


def slon(xK, yK, xS, yS):
    return abs(xK - xS) == abs(yK - yS)


def vse(xK, yK, xL, yL, xS, yS):
    return lagya(xK, yK, xL, yL) and slon(xK, yK, xS, yS)

print('введите кооридинаты короля: ')
xK = int(input())
yK = int(input())
print('введите кооридинаты ладьи: ')
xL = int(input())
yL = int(input())
print('введите кооридинаты слона: ')
xS = int(input())
yS = int(input())

if vse(xK, yK, xL, yL, xS, yS):
    print('королю угрожают слон и ладья')
elif lagya(xK, yK, xL, yL):
    print('королю угрожает ладья ')
elif slon(xK, yK, xS, yS):
    print('королю угрожает слон ')
else:
    print('король не находится под угрозой')

File: dynamic_lib/chess.cpp
#include <iostream>
#include "libchess.h"
using namespace std;

int main() {
    cout << "Введите координаты короля: " << endl;
    int xK, yK;
    cin >> xK >> yK;

    cout << "Введите координаты ладьи: " << endl;
    int xL, yL;
    cin >> xL >> yL;

    cout << "Введите координаты слона: " << endl;
    int xS, yS;
    cin >> xS >> yS;

    if (vse(xK, yK, xL, yL, xS, yS)) {
        cout << "Королю угрожают слон и ладья" << endl;
    } else if (lagya(xK, yK, xL, yL)) {
        cout << "Королю угрожает ладья" << endl;
    } else if (slon(xK, yK, xS, yS)) {
        cout << "Королю угрожает слон" << endl;
    } else {
        cout << "Король не находится под угрозой" << endl;
    }

    return 0;
}
